packages/fxa-settings/src/pages/Pair/Auth/Allow/index.test.tsx

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, screen } from '@testing-library/react';
+import AuthAllow, { viewName } from '.';
+import {
+  MOCK_METADATA_UNKNOWN_LOCATION,
+  MOCK_METADATA_WITH_DEVICE_NAME,
+  MOCK_METADATA_WITH_LOCATION,
+} from '../../../../components/DeviceInfoBlock/mocks';
+import { MOCK_ACCOUNT } from '../../../../models/mocks';
+import { usePageViewEvent } from '../../../../lib/metrics';
+import { REACT_ENTRYPOINT } from '../../../../constants';
+
+jest.mock('../../../../lib/metrics', () => ({
+  usePageViewEvent: jest.fn(),
+  logViewEvent: jest.fn(),
+}));
+
+describe('Pair/Supp/Allow page', () => {
+  it('renders as expected when the location is undefined', () => {
+    render(
+      <AuthAllow
+        email={MOCK_ACCOUNT.primaryEmail.email}
+        authDeviceInfo={MOCK_METADATA_UNKNOWN_LOCATION}
+      />
+    );
+
+    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent(
+      'Did you just sign in to Firefox?'
+    );
+    screen.getByText('johndope@example.com');
+    screen.getByText('Firefox on macOS');
+    screen.getByText('Location unknown');
+    screen.getByText('IP address: XX.XX.XXX.XXX');
+  });
+
+  it('renders as expected when a device name is provided', () => {
+    render(
+      <AuthAllow
+        email={MOCK_ACCOUNT.primaryEmail.email}
+        authDeviceInfo={MOCK_METADATA_WITH_DEVICE_NAME}
+      />
+    );
+
+    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
+      'Ultron'
+    );
+  });
+
+  it('renders as expected when a location is available', () => {
+    render(
+      <AuthAllow
+        email={MOCK_ACCOUNT.primaryEmail.email}
+        authDeviceInfo={MOCK_METADATA_WITH_LOCATION}
+      />
+    );
+
+    screen.getByText('Vancouver, British Columbia, Canada (estimated)');
+  });
+
+  it('emits the expected metrics event on render', () => {
+    render(
+      <AuthAllow
+        email={MOCK_ACCOUNT.primaryEmail.email}
+        authDeviceInfo={MOCK_METADATA_WITH_DEVICE_NAME}
+      />
+    );
+
+    expect(usePageViewEvent).toHaveBeenCalledWith(
+      'pair.auth.allow',
```

Comment:
Q: Should this use the exported `export const viewName = 'pair.auth.allow';` from `packages/fxa-settings/src/pages/Pair/Auth/Allow/index.tsx#L29` below?